script/reference-device/dhcp6_pd_daemon.py

```diff
@@ -0,0 +1,115 @@
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import logging
+import dbus
+import gi.repository.GLib as GLib
+import subprocess
+import threading
+import os
+
+from dbus.mainloop.glib import DBusGMainLoop
+DBusGMainLoop(set_as_default=True)
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+bus = dbus.SystemBus()
+thread_dbus_obj = None
+iface = None
+intended_dhcp6pd_state = None
+
+def restart_dhcpcd_with_custom_config():
+    global intended_dhcp6pd_state
+    if os.path.isfile("/etc/dhcpcd.conf.custom"):
+        try:
+            subprocess.run(["sudo", "cp", "/etc/dhcpcd.conf.custom", "/etc/dhcpcd.conf"], check=True)
+            subprocess.run(["sudo", "systemctl", "daemon-reload"], check=True)
+            subprocess.run(["sudo", "service", "dhcpcd", "restart"], check=True)
+            logging.info("Successfully restarted dhcpcd service.")
+            intended_dhcp6pd_state = None
+        except subprocess.CalledProcessError as e:
+            logging.error(f"Error restarting dhcpcd service: {e}")
+            intended_dhcp6pd_state = None
+    else:
+        logging.error("Customised configuration file not found. Cannot apply custom.")
+        intended_dhcp6pd_state = None
+
+def restore_default_config():
+    if os.path.isfile("/etc/dhcpcd.conf.orig"):
+        try:
+            subprocess.run(["sudo", "cp", "/etc/dhcpcd.conf.orig", "/etc/dhcpcd.conf"], check=True)
+            subprocess.run(["sudo", "systemctl", "daemon-reload"], check=True)
+            subprocess.run(["sudo", "service", "dhcpcd", "restart"], check=True)
+            logging.info("Successfully restored default dhcpcd config.")
+        except subprocess.CalledProcessError as e:
+            logging.error(f"Error restoring dhcpcd config: {e}")
+    else:
+        logging.error("Backup configuration file not found. Cannot restore default.")
+
+def properties_changed_handler(interface_name, changed_properties, invalidated_properties):
+    global intended_dhcp6pd_state
+    if "Dhcp6PdState" in changed_properties:
+        new_state = changed_properties["Dhcp6PdState"]
+        if new_state == "running" and intended_dhcp6pd_state != "running":
+            logging.info(f"Dhcp6PdState changed to: {new_state}")
+            intended_dhcp6pd_state = "running"
+            thread = threading.Thread(target=restart_dhcpcd_with_custom_config)
+            thread.start()
+        elif new_state in ("stopped", "disabled") and intended_dhcp6pd_state is None:
+            logging.info(f"Dhcp6PdState changed to: {new_state}")
+            restore_default_config()
+
+def connect_to_signal():
+    global thread_dbus_obj, thread_properties_dbus_iface
```

Comment:
Generally I'd suggest to avoid global variables as much as possible. Could you achieve it by returning a tuple `(thread_dbus_obj, thread_properties_dbus_iface)`?